File: DevExpress-Demo/CS/DiagramDemo.Wpf/Modules/EventsExamples/MaxWidthExample.cs
using System;
using System.Linq;
using System.Windows;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Diagram;
using ResizeMode = DevExpress.Diagram.Core.ResizeMode;

namespace DiagramDemo {
    public static class MaxWidthExample {
        public static FrameworkElement Run() {
            var diagram = new DiagramControl() { PageSize = new Size(400, 300), EnableProportionalResizing = false };
            diagram.Loaded += (s, e) => diagram.FitToPage();
            // Code Example Start
            // Each time the end-user tries to resize a diagram item, the ItemsResizing event is raised.
            // The following implementation sets the maximum width based on the item's content.

            diagram.ItemsResizing += (sender, e) => {
                foreach(ResizingItem c in e.Items) {
                    double maxWidth = 0d;
                    if(c.Item is DiagramShape)
                        double.TryParse(((DiagramShape)c.Item).Content, out maxWidth);
                    double widthOver = c.NewSize.Width - maxWidth;
                    if(widthOver <= 0d) continue;

                    c.NewSize = new Size(maxWidth, c.NewSize.Height);
                    if(e.Mode == ResizeMode.Left || e.Mode == ResizeMode.TopLeft || e.Mode == ResizeMode.BottomLeft)
                        c.NewDiagramPosition = new Point(c.NewDiagramPosition.X + widthOver, c.NewDiagramPosition.Y);
                }
            };

            diagram.Items.Add(new DiagramShape() { Content = "300", Position = new Point(60, 75), Width = 280, Height = 150, CanRotate = false, FontSize = 24 });
            // Code Example End
            return diagram;
        }
    }
}

File: DevExpress-Demo/VB/DiagramDemo.Wpf/Modules/EventsExamples/MaxWidthExample.vb
Imports System.Windows
Imports DevExpress.Xpf.Diagram
Imports ResizeMode = DevExpress.Diagram.Core.ResizeMode

Namespace DiagramDemo

    Public Module MaxWidthExample

        Public Function Run() As FrameworkElement
            Dim diagram = New DiagramControl() With {.PageSize = New Size(400, 300), .EnableProportionalResizing = False}
            AddHandler diagram.Loaded, Sub(s, e) diagram.FitToPage()
            ' Code Example Start
            ' Each time the end-user tries to resize a diagram item, the ItemsResizing event is raised.
            ' The following implementation sets the maximum width based on the item's content.
            AddHandler diagram.ItemsResizing, Sub(sender, e)
                For Each c As ResizingItem In e.Items
                    Dim maxWidth As Double = 0R
                    If TypeOf c.Item Is DiagramShape Then Call Double.TryParse(CType(c.Item, DiagramShape).Content, maxWidth)
                    Dim widthOver As Double = c.NewSize.Width - maxWidth
                    If widthOver <= 0R Then Continue For
                    c.NewSize = New Size(maxWidth, c.NewSize.Height)
                    If e.Mode = ResizeMode.Left OrElse e.Mode = ResizeMode.TopLeft OrElse e.Mode = ResizeMode.BottomLeft Then c.NewDiagramPosition = New Point(c.NewDiagramPosition.X + widthOver, c.NewDiagramPosition.Y)
                Next
            End Sub
            diagram.Items.Add(New DiagramShape() With {.Content = "300", .Position = New Point(60, 75), .Width = 280, .Height = 150, .CanRotate = False, .FontSize = 24})
            ' Code Example End
            Return diagram
        End Function
    End Module
End Namespace
